source/common/upstream/upstream_impl.h

```diff
@@ -321,7 +321,7 @@ class StrictDnsClusterImpl : public BaseDynamicClusterImpl {
 public:
   StrictDnsClusterImpl(const Json::Object& config, Runtime::Loader& runtime, Stats::Store& stats,
                        Ssl::ContextManager& ssl_context_manager, Network::DnsResolver& dns_resolver,
-                       Event::Dispatcher& dispatcher);
+                       Network::DnsResolverPtr custom_dns_resolver, Event::Dispatcher& dispatcher);
```

Comment:
Maybe we could simplify this by using a `DnsResolverSharedPtr` and passing in the normal resolver using this as well? That way we only have one `DnsResolver` argument.